cardano-api/src/Cardano/Api/SerialiseRaw.hs

```diff
@@ -32,28 +33,32 @@ serialiseToRawBytesHex = Base16.encode . serialiseToRawBytes
 serialiseToRawBytesHexText :: SerialiseAsRawBytes a => a -> Text
 serialiseToRawBytesHexText = Text.decodeUtf8 . serialiseToRawBytesHex
 
+-- | The errors that the pure 'SerialiseAsRawBytes' parsing\/decoding functions can return.
 data RawBytesHexError
   = RawBytesHexErrorBase16DecodeFail
       ByteString -- ^ original input
       String -- ^ error message
   | RawBytesHexErrorRawBytesDecodeFail
       ByteString -- ^ original input
-      -- TODO(2022-01-26, cblp) TypeRep -- ^ output type proxy
+      TypeRep    -- ^ expected type
   deriving (Show)
 
 instance Error RawBytesHexError where
   displayError = \case
     RawBytesHexErrorBase16DecodeFail input message ->
-      "Expected Base16-encoded bytestring, but got " ++ show input ++ "; "
+      "Expected Base16-encoded bytestring, but got " ++ pretty input ++ "; "
       ++ message
-    RawBytesHexErrorRawBytesDecodeFail input ->
-      "Failed to deserialise " ++ show input
-      -- TODO(2022-01-26, cblp) show expected output type
+    RawBytesHexErrorRawBytesDecodeFail input asType ->
+      "Failed to deserialise " ++ pretty input ++ " as " ++ (show asType)
+    where
+      pretty bs = case Text.decodeUtf8' bs of
+        Right t -> Text.unpack t
+        Left _ -> show bs
```

Comment:
Let's also include the error from `decodeUtf8'` e.g "Error decoding: " ++ bs ++ " Error: " ++ decodeErr

Reply:
For me it would be confusing to see an error about UTF8 decoding.
The no-error-case does not even try UTF8 decoding and a correct input is hex-digits only.

Reply:
> For me it would be confusing to see an error about UTF8 decoding.

It would inform the user that what they are trying to decode is not UTF8 encoding. Why is that a bad thing/confusing?